File: 3 - Linguagem C++/3 - Vetor - ler-nomes-e-idades-mostrar-pessoa-mais-velha.cpp
#include <iostream>
#include <iomanip>
#include <climits>

using namespace std;

int main()
{
   int N, posMaior, maior;

   cout << "Quantas pessoas voce vai digitar? ";
   cin >> N;

   string nomes[N];
   int idades[N];

   for (int i=0; i<N; i++){
      cout << "Dados da" << i + 1 << "a pessoa: " << endl;
      cout << "NOME: ";
      cin.ignore(INT_MAX, '\n');
      getline(cin, nomes[i]);
      cout << "IDADE: ";
      cin >> idades[i];
     }

   posMaior = 0;
   maior = idades[0];

   for (int i=0; i<N; i++){
       if (idades[i] > maior){
         maior = idades[i];
         posMaior = i;
       }
   }

       cout << fixed << setprecision(1);
       cout << "PESSOA MAIS VELHA = " << nomes[posMaior] << endl;

   return 0;
}

File: 4 - JAVA/3 - Vetor - ler-nomes-e-idades-mostrar-pessoa-mais-velha.java
import java.util.Locale;
import java.util.Scanner;

public class Main {
	
	public static void main(String[] args) {
		
		Locale.setDefault(Locale.US);
		Scanner sc = new Scanner(System.in);
		
		int N, i, maior, posMaior;
		
		System.out.print("Quantas pessoas voce vai digitar? ");
		N = sc.nextInt(); 
		
		int[] idades = new int[N];
		String[] nomes = new String[N];
		
		for (i=0; i<N; i++) {
			System.out.printf("Dados da %da pessoa:\n", i + 1);
			System.out.print("Nome: ");  
			nomes[i] = sc.next();
			System.out.print("Idade: ");
			idades[i] = sc.nextInt();
		}
		
		maior = idades[0];
		posMaior = 0;
		for (i=1; i<N; i++) {
			if (idades[i] > maior) {
				maior = idades[i];
	            posMaior = i;
			}
		}
		System.out.print("PESSOA MAIS VELHA: " + nomes[posMaior]);
		
		sc.close();
	}
}